File: java/src/main/java/com/experian/dvs/client/common/ResponseError.java
package com.experian.dvs.client.common;

import com.experian.dvs.client.server.RestApiResponseError;

import java.util.Objects;

public class ResponseError {

    private final String type;
    private final String title;
    private final String detail;
    private final String instance;

    public ResponseError(final RestApiResponseError error) {
        this.type = Objects.toString(error.getType(), "");
        this.title = Objects.toString(error.getTitle(), "");
        this.detail = Objects.toString(error.getDetail(), "");
        this.instance = Objects.toString(error.getInstance(), "");
    }


    public String getType() {
        return type;
    }

    public String getTitle() {
        return title;
    }

    public String getDetail() {
        return detail;
    }

    public String getInstance() {
        return instance;
    }
}

File: csharp/DVSClient/DVSClient/common/ResponseError.cs
using DVSClient.Server;

namespace DVSClient.Common
{
    public class ResponseError
    {
        public string Type { get; }
        public string Title { get; }
        public string Detail { get; }
        public string Instance { get; }

        public ResponseError(RestApiResponseError error)
        {
            Type = error?.Type ?? string.Empty;
            Title = error?.Title ?? string.Empty;
            Detail = error?.Detail ?? string.Empty;
            Instance = error?.Instance ?? string.Empty;
        }
    }
}
